### DISCUSSION TITLE
3.0.0 compile error

### DISCUSSION THREAD

**Discussion by gvanyao** [apache/pulsar#20815] [Q&A]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project buildtools: Compilation failure
[ERROR] /Users/yaogvan/IdeaProjects/pulsar3.0/buildtools/src/main/java/org/apache/pulsar/tests/BetweenTestClassesListenerAdapter.java:[21,18] cannot access org.testng.IClassListener
[ERROR]   bad class file: /usr/local/apache-maven-3.8.6/localRepository/org/testng/testng/7.7.1/testng-7.7.1.jar(org/testng/IClassListener.class)
[ERROR]     class file has wrong version 55.0, should be 52.0
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

**Comment by tisonkun** [comment]
What's your JDK version and Maven version?

> class file has wrong version 55.0, should be 52.0

You can try to delete `/usr/local/apache-maven-3.8.6/localRepository/org/testng/testng` and rerun the command, or delete the whole maven local repository and rerun with JDK 17.
